packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -524,7 +524,10 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     WidgetsBinding.instance.addPostFrameCallback((_) {
       final BuildContext? highlightContext = buttonItemKeys[currentHighlight!].currentContext;
       if (highlightContext != null) {
-        Scrollable.ensureVisible(highlightContext);
```

Comment:
shouldn't we fix `Scrollable.ensureVisible`? Surely it's a bug that calling that doesn't work.

Reply:
DropdownMenu` calls `Scrollable.ensureVisible` which then calls `ScrollPosition.ensureVisible`. However, after the `OverlayPortal` migration, it seems the correct scrollable context isn't passed in nested scroll views.

https://github.com/flutter/flutter/blob/1465da40b7c1899ce06fa55ff249ffc046182461/packages/flutter/lib/src/widgets/scrollable.dart#L471-L476

https://github.com/flutter/flutter/blob/1465da40b7c1899ce06fa55ff249ffc046182461/packages/flutter/lib/src/widgets/scrollable.dart#L1028

My fix finds the scrollable and calls `ScrollPosition.ensureVisible` directly. 

 I'm not expert in scrollables so I'll defer to @Piinks, if this makes sense.

Reply:
I can reproduce the issue in the official `OverlayPortal` example by nesting scroll views. 

Hixie is right it is a bug in `Scrollable.ensureVisible`.  However, I'm not sure `ensureVisible` can access `ScrollableState` outside `OverlayPortal` context. 

### Code sample

<details>
<summary>expand to view the code sample</summary> 

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [OverlayPortal].

void main() => runApp(const OverlayPortalExampleApp());

class OverlayPortalExampleApp extends StatelessWidget {
  const OverlayPortalExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('OverlayPortal Example')),
        body: const Center(child: ClickableTooltipWidget()),
      ),
    );
  }
}

class ClickableTooltipWidget extends StatefulWidget {
  const ClickableTooltipWidget({super.key});

  @override
  State<StatefulWidget> createState() => ClickableTooltipWidgetState();
}

class ClickableTooltipWidgetState extends State<ClickableTooltipWidget> {
  final OverlayPortalController _tooltipController = OverlayPortalController();

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      child: Card(
        child: SingleChildScrollView(
          child: TextButton(
            onPressed: _tooltipController.toggle,
            child: DefaultTextStyle(
              style: DefaultTextStyle.of(context).style.copyWith(fontSize: 50),
              child: OverlayPortal(
                controller: _tooltipController,
                overlayChildBuilder: (BuildContext context) {
                  return  Positioned(
                    right: 50,
                    bottom: 50,
                    child: GestureDetector(
                      onTap: () {
                        print('tooltip tapped');
                        Scrollable.ensureVisible(context);
                      },
                      child: const ColoredBox(
                        color: Colors.amberAccent,
                        child: Text('tooltip'),
                      ),
                    ),
                  );
                },
                child: const Text('Press to show/hide tooltip'),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

Reply:
https://github.com/flutter/flutter/pull/67773 made similar fix for nested scroll views, it doesn't handle `OverlayPortal` scenario very well and this is where the error is coming.

I'll close this PR to find out why this fails in an `OverlayPortal` and fix it `ensureVisible` instead`.